kstreams/types.py

```diff
@@ -23,9 +27,17 @@ def __call__(
         headers: typing.Optional[Headers] = None,
         serializer: typing.Optional["Serializer"] = None,
         serializer_kwargs: typing.Optional[typing.Dict] = None,
+        producer: typing.Optional[Producer] = None,
     ) -> typing.Awaitable[RecordMetadata]: ...
 
 
+class Transaction(typing.Protocol):
```

Comment:
Add a docstring to this class please. It's not clear why we have a protocol `Transaction` and then we have `transaction.Transaction`. An explanation on the docstring would clarify.

consider also renaming this to TransactionProtocol.